internal/controller/storageclient_controller.go

```diff
@@ -309,6 +313,47 @@ func (r *StorageClientReconciler) reconcilePhases() (ctrl.Result, error) {
 			if err != nil {
 				return reconcile.Result{}, fmt.Errorf("failed to create remote noobaa: %v", err)
 			}
+		case "ServiceSpec":
+			serviceSpec := &v1.ServiceSpec{}
+			if err := json.Unmarshal(eResource.Data, &serviceSpec); err != nil {
+				return reconcile.Result{}, fmt.Errorf("failed to unmarshall service spec data: %v", err)
+			}
+			srv := &v1.Service{}
+			srv.Name = eResource.Name
+			srv.Namespace = r.OperatorNamespace
+
+			_, err = controllerutil.CreateOrUpdate(r.ctx, r.Client, srv, func() error {
+				if err := r.own(srv); err != nil {
+					return err
+				}
+
+				srv.Spec = *serviceSpec
+				return nil
+			})
+			if err != nil {
+				return reconcile.Result{}, fmt.Errorf("failed to create service: %v", err)
+			}
+		case "Endpoints":
```

Comment:
I see in provider side this is `EndpointSubset`?

Reply:
I was trying it out, but shifted back to Endpoints in latest commit